File: tests/examples_tests/python/wrappers/overloads/overload_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "examples/cxx/overloads/overload.hpp"

namespace py = pybind11;

void bindOverloadedFunctions(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::OverloadedFunctions, std::shared_ptr<iegen::example::OverloadedFunctions>> overloadedfunctions(m, "OverloadedFunctions");

    overloadedfunctions.def(py::init<>());
    overloadedfunctions.def("concatenate", py::overload_cast<std::string, std::string>(&iegen::example::OverloadedFunctions::concatenate), py::arg("first"), py::arg("second"));
    overloadedfunctions.def("concatenate", py::overload_cast<std::string, std::string, std::string>(&iegen::example::OverloadedFunctions::concatenate), py::arg("first"), py::arg("second"), py::arg("third"));
}

File: tests/examples_tests/python/src/overloads/overload_pygen.py
from __future__ import annotations
from typing import *
import iegen.examples.overloads.overload as pybind_overload_pygen
from bind_utils import *
from metaclass import *


class Overload(metaclass=OriginalMethodsMetaclass):
    """
    An example for with overloaded methods."""
    @bind
    def __init__(self):
        pass


    
    @bind
    def concatenate(self, first: str, second: str) -> str:
        """
        Concatenate with two strings.
        """
        pass
    @bind
    def concatenate(self, first: str, second: str, third: str) -> str:
        """
        Concatenate with three strings.
        """
        pass

